src/dswx_sar/save_mgrs_tiles.py

```diff
@@ -88,41 +127,179 @@ def save_mgrs_tile(source_tif_path,
+                           yRes=yspacing,
+                           outputBounds=bbox,
+                           resampleAlg=method,
+                           format='GTIFF')
     gdal.Warp(output_tif_file_path, source_tif_path, options=opt)
-    dswx_sar_util._save_as_cog(output_tif_file_path, output_dir_path, logger,
-                compression='ZSTD',
-                nbits=16)
+    input_tif_obj = None
+    dswx_sar_util._save_as_cog(output_tif_file_path,
+                               output_dir_path,
+                               logger,
+                               compression='ZSTD',
+                               nbits=16)
+
+
+def find_mgrs_collection(image_tif,
+                         mgrs_collection_file=None):
+    """Find MGRS tile collection intersecting Geotiff
+    Parameters
+    ----------
+    image_tif: str
+        path of input geotiff
+    mgrs_collection_file : str
+        path for MGRS tile collection
+
+    Returns
+    ----------
+    mgrs_list: list
+        list of MGRS tiles
+    """
+    if mgrs_collection_file is not None:
+        # Load the raster data
+        with rasterio.open(image_tif) as src:
+            epsg_code = int(src.crs.data['init'].split(':')[1])
+
+            # Get bounds of the raster data
+            left, bottom, right, top = src.bounds
+
+            # Reproject to EPSG 4326 if the current EPSG is not 4326
+            if epsg_code != 4326:
+                left, bottom, right, top = transform_bounds(
+                                                    src.crs,
+                                                    {'init': 'EPSG:4326'},
+                                                    left,
+                                                    bottom,
+                                                    right,
+                                                    top)
+
+        # Create a Polygon from the bounds
+        raster_polygon = Polygon([(left, bottom),
+                                  (left, top),
+                                  (right, top),
+                                  (right, bottom)])
+
+        # Create a GeoDataFrame from the raster polygon
+        raster_gdf = gpd.GeoDataFrame([1],
+                                      geometry=[raster_polygon],
+                                      crs={'init': 'EPSG:4326'})
+
+        # Load the vector data
+        vector_gdf = gpd.read_file(mgrs_collection_file)
+
+        # Calculate the intersection
+        intersection = gpd.overlay(raster_gdf,
+                                   vector_gdf,
+                                   how='intersection')
+
+        # Add a new column with the intersection area
+        intersection['Area'] = intersection.geometry.area
+
+        # Find the polygon with the maximum intersection area
+        most_overlapped = intersection.loc[intersection['Area'].idxmax()]
+
+        mgrs_list = ast.literal_eval(most_overlapped['mgrs_tiles'])
+
+    else:
```

Comment:
Please consider separating this function in two, one for the case in which the user provides the MGRS tile collection and another for the the collection is not provided.

Reply:
Yes. Thank you for the comments. This function was separated into `get_intersecting_mgrs_tiles_list_from_db` and `get_intersecting_mgrs_tiles_list`